docker/data-science/notebooks/aips.py

```diff
@@ -285,9 +285,39 @@ def img_path_for_upc(upc):
     return expected_jpg_path if os.path.exists(expected_jpg_path) else unavailable_jpg_path
 
 def display_search(query, documents):
-  display(HTML(f"<strong>Query</strong>: <i>{query}</i><br/><br/><strong>Results:</strong>"))
-  display(HTML(documents))
-  
+    display(HTML(f"<strong>Query</strong>: <i>{query}</i><br/><br/><strong>Results:</strong>"))
+    display(HTML(documents))
+
+def display_product_search(query, documents):
+    file_path = os.path.dirname(os.path.abspath(__file__))
+    search_results_template_file = os.path.join(file_path + "/data/retrotech/templates/", "search-results.html")
+    with open(search_results_template_file) as file:
+        file_content = file.read()
+
+        template_syntax = "<!-- BEGIN_TEMPLATE[^>]*-->(.*)<!-- END_TEMPLATE[^>]*-->"
+        header_template = re.sub(template_syntax, "", file_content, flags=re.S)
+
+        results_template_syntax = "<!-- BEGIN_TEMPLATE: SEARCH_RESULTS -->(.*)<!-- END_TEMPLATE: SEARCH_RESULTS[^>]*-->"
+        x = re.search(results_template_syntax, file_content, flags=re.S)
+        results_template = x.group(1)
+
+        separator_template_syntax = "<!-- BEGIN_TEMPLATE: SEPARATOR -->(.*)<!-- END_TEMPLATE: SEPARATOR[^>]*-->"
+        x = re.search(separator_template_syntax, file_content, flags=re.S)
+        separator_template = x.group(1)
+
+        rendered = header_template.replace("${QUERY}", query)
```

Comment:
We can also use https://docs.python.org/3.4/library/string.html#template-strings to replace templates

Reply:
I'll do a swift refactor on this templating logic when the major chapter refactor is finished, these changes were marked when I fixed tabulation